paymentsheet/src/main/java/com/stripe/android/paymentsheet/PaymentSheet.kt

```diff
@@ -1321,8 +1329,35 @@ class PaymentSheet internal constructor(
         /**
          * A short-lived token that allows the SDK to access a Customer's payment methods.
          */
-        val ephemeralKeySecret: String
-    ) : Parcelable
+        val ephemeralKeySecret: String,
+
+        internal val accessType: CustomerAccessType,
+    ) : Parcelable {
+        constructor(
+            id: String,
+            ephemeralKeySecret: String,
+        ) : this(
+            id = id,
+            ephemeralKeySecret = ephemeralKeySecret,
+            accessType = CustomerAccessType.LegacyCustomerEphemeralKey(ephemeralKeySecret)
+        )
+
+        @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+        companion object {
+            @ExperimentalCustomerSessionApi
+            @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+            fun createWithCustomerSession(
+                id: String,
+                customerSessionClientSecret: String
```

Comment:
```suggestion
                clientSecret: String
```

Is what API review had.